Explorer/Assets/DCL/Multiplayer/Movement/Systems/PlayerMovementNetSendSystem.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.Character.Components;
+using DCL.CharacterMotion.Components;
+using DCL.Diagnostics;
+using DCL.Multiplayer.Connections.RoomHubs;
+using DCL.Multiplayer.Movement.Settings;
+using DCL.Multiplayer.Movement.System;
+using ECS.Abstract;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace DCL.Multiplayer.Movement.Systems
+{
+    [UpdateInGroup(typeof(PostRenderingSystemGroup))]
+    [LogCategory(ReportCategory.MULTIPLAYER_MOVEMENT)]
+    public partial class PlayerMovementNetSendSystem : BaseUnityLoopSystem
+    {
+        private const int MAX_MESSAGES_PER_SEC = 10; // 10 Hz == 10 [msg/sec]
+
+        private readonly MultiplayerMovementMessageBus messageBus;
+        private readonly IMultiplayerMovementSettings settings;
+
+        private readonly CharacterController playerCharacter;
+
+        private FullMovementMessage? lastSentMessage;
+
+        private int messagesSentInSec;
+        private float mesPerSecResetCooldown;
+
+        public PlayerMovementNetSendSystem(World world, MultiplayerMovementMessageBus messageBus, IMultiplayerMovementSettings settings, CharacterController playerCharacter) : base(world)
+        {
+            this.messageBus = messageBus;
+            this.settings = settings;
+            this.playerCharacter = playerCharacter;
+        }
+
+        protected override void Update(float t)
+        {
+            UpdateMessagePerSecondCounter(t);
+            SendPlayerNetMovementQuery(World);
+        }
+
+        private void UpdateMessagePerSecondCounter(float t)
+        {
+            if (mesPerSecResetCooldown > 0)
+                mesPerSecResetCooldown -= t;
+            else
+            {
+                // Debug.Log($"VVV ------- MES PER SEC: <color={GetColorBasedOnMesPerSec(MessagesSentInSec)}> {MessagesSentInSec} </color> ----------");
+                mesPerSecResetCooldown = 1; // 1 [sec]
+                messagesSentInSec = 0;
+            }
+        }
+
+        [Query]
+        [All(typeof(PlayerComponent))]
+        private void SendPlayerNetMovement(ref CharacterAnimationComponent animation, ref StunComponent stun, ref MovementInputComponent move, ref JumpInputComponent jump)
+        {
+            if (messagesSentInSec >= MAX_MESSAGES_PER_SEC) return;
+
+            if (lastSentMessage == null)
```

Comment:
A warning: you iterate over entities but presume `lastSentMessage` exists in a single entity always. Not great stylistically